lib/logstorage/datadb.go

```diff
@@ -321,15 +341,21 @@ func assertIsInMerge(pws []*partWrapper) {
 	}
 }
 
-// mustMergeParts merges pws to a single resulting part.
+var errReadOnly = errors.New("the storage is in read-only mode")
+
+// mergeParts merges pws to a single resulting part.
 //
 // if isFinal is set, then the resulting part will be saved to disk.
 //
 // All the parts inside pws must have isInMerge field set to true.
-func (ddb *datadb) mustMergeParts(pws []*partWrapper, isFinal bool) {
+func (ddb *datadb) mergeParts(pws []*partWrapper, isFinal bool) error {
```

Comment:
I don't see any errors produced by this function.

Reply:
it looks like mustMergeBlockStreams must produce an error if there is no enough disk space during merge process.

Reply:
Atm, it produces an error here:
```golang
if ddb.IsReadOnly() {
    return errReadOnly
}
```

> it looks like mustMergeBlockStreams must produce an error if there is no enough disk space during merge process.

Sounds good to me, but this PR is quite large already, and I'd wanted to avoid adding something else here.
This one is kind of starting point to have a better error handling for `logstorage/` as it mostly uses `must*` without error handling.
Wdyt about addressing this in a follow-up PR?

Reply:
Sounds good